### ISSUE TITLE
Using WebApp.Start for hosting IdentityServer results in assets loading issues (runAllManagedModulesForAllRequests  doesn't apply here)

### ISSUE THREAD

**Issue by jorgecavaleiro** [IdentityServer/IdentityServer3#2773]
- [x] I read and understood how to enable [logging](https://identityserver.github.io/Documentation/docsv2/configuration/logging.html)

### Question / Issue

I'm trying to use the simple self-hosting technique to host IdentityServer for development environments (the '**runAllManagedModulesForAllRequests** ' is an IIS setting, so it's not applicable here). 

The service runs well but when the user is redirected to the login page, the assets requests returns **404**:
Request URL:https://localhost:44333/core/**assets**/scripts.2.5.0.js
Request Method:GET
Status Code:404 
Remote Address:[::1]:44333

Any help appreciated. 

### Relevant parts of the log file

```
<log goes here>
```

**Comment by brockallen** [comment]
What are you using for hosting? Katana self-host?

**Comment by jorgecavaleiro** [comment]
Yes.
Like this:

var webApp = WebApp.Start(ConfigurationManager.AppSettings[IdentityConstants.AppSetting.AuthorityEndpoint], IdSrvApp =>
            {
                IdSrvApp.UseIdentityServer();
            });

**Comment by jorgecavaleiro** [comment]
Were trying to embed the identity server in our C# solutions just for development purpose (in production the identity server runs in isolated web site).
I wonder what should be the best strategy to accomplish this..
This current issue  and the #2772, are related to two different approaches I tried in order to accomplish this goal.

**Comment by leastprivilege** [comment]
You need to add the static files middleware

**Comment by jorgecavaleiro** [comment]
We decided for a different approach.
For our development environments were going to launch a console application with the identity server self-hosted.
We think that will probably less tricky than hosting it side by side with our web application.

Thank both.
I'm going to close this.

**Comment by parkinsona** [comment]
I have just upgraded from v2.3 to v2.5 and have just got an issue very similar to this.
It happens when the user is redirected to the login screen.
I get the following in my log:

``` 
Exception Message: The controller for path '/core/assets/scripts.2.0.0.js' was not found or does not implement IController.

Stack Trace: 
   at System.Web.Mvc.DefaultControllerFactory.GetControllerInstance(RequestContext requestContext, Type controllerType)
   at System.Web.Mvc.DefaultControllerFactory.CreateController(RequestContext requestContext, String controllerName)
```

Curious that this issue was raised just a few days ago after 2.5 was released?

**Comment by leastprivilege** [comment]
Try our self-host in the core repo and see if you can repro that.

**Comment by parkinsona** [comment]
I've just checked my TEST environment (v2.3) and it also has this kind of issue. Different resource though: 
``` 
Exception Message: The controller for path '/core/assets/angular.min.js.map' was not found or does not implement IController. 
```

I'll look at trying to repro in a bit.

**Comment by parkinsona** [comment]
I still haven't had a chance to try and repro this, but have found it is also happening with:

```
Exception Message: The controller for path '/favicon.ico' was not found or does not implement IController.
```

**Comment by parkinsona** [comment]
@leastprivilege , I have been able to repro this using the MVC ViewService Sample.
I can reproduce the issue from chrome when it tries to access favicon.ico.
If I go to https://localhost:44333/core/.well-known/openid-configuration from a fresh clear of chrome's favicons, the it will raise an application error complaining about the /favicon.ico

I can get around the favicon issue by adding it in myself. But I'm not so sure about the other assets.

**Comment by leastprivilege** [comment]
Please use on of the samles that use the default view service.

**Comment by parkinsona** [comment]
I've tried the MVC Authentication, and it worked correctly although it did includ a /favicon.ico.
So I included a favicon.ico to my project, and  that seems to have cleared up my issue in 2.3.

I also found the issue with v2.5. I am using a custom layout.html file, and I was referencing 
{applicationPath}/assets/scripts.2.0.0.js This must have been included in v2.3 but not in v2.5. I have updated to v2.5 and that's sorted that out.

The issue I was getting in TEST, was a caching issue. On another machine it didn't throw the error. I cleared my cache and had to close firebug for it to fix it in my browser.

So all good from here.
Question though... When there's a new version of identity server, will I have to manually adjust my cutsom layout.html with the correct version of the scripts js?
